src/NodaTime/Text/AnnualDatePattern.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2018 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using System.Globalization;
+using System.Text;
+using NodaTime.Annotations;
+using NodaTime.Globalization;
+using NodaTime.Text.Patterns;
+using NodaTime.Utility;
+using JetBrains.Annotations;
+
+namespace NodaTime.Text
+{
+    /// <summary>
+    /// Represents a pattern for parsing and formatting <see cref="AnnualDate"/> values.
+    /// </summary>
+    /// <threadsafety>
+    /// When used with a read-only <see cref="CultureInfo" />, this type is immutable and instances
+    /// may be shared freely between threads. We recommend only using read-only cultures for patterns, although this is
+    /// not currently enforced.
+    /// </threadsafety>
+    [Immutable] // Well, assuming an immutable culture...
+    public sealed class AnnualDatePattern : IPattern<AnnualDate>
+    {
+        internal static readonly AnnualDate DefaultTemplateValue = new AnnualDate(1, 1);
+
+        private const string DefaultFormatPattern = "G"; // General, ISO-like
+
+        internal static readonly PatternBclSupport<AnnualDate> BclSupport =
+            new PatternBclSupport<AnnualDate>(DefaultFormatPattern, fi => fi.AnnualDatePatternParser);
+
+        /// <summary>
+        /// Gets an invariant local date pattern which is compatible with the month/day part of ISO-8601.
+        /// This corresponds to the text pattern "MM'-'dd".
```

Comment:
Note that `javax.time.MonthDay` uses `--MM-dd` (for the debugging-oriented `toString()`), presumably to avoid confusion with their `YearMonth`. We don't have the latter, and even if we did, I don't think we need need our default patterns to be non-overlapping.

(Surprisingly, ISO 8601 appears to have nothing to say about formatting a month and day, even in the [imo ridiculous] recent update draft.)